src/otaclient_common/linux.py

```diff
@@ -302,3 +303,34 @@ def fstrim_at_subprocess(
         stderr=subprocess.DEVNULL,
         shell=True,
     )
+
+
+def is_regular_file(path: Path) -> bool:
+    """Check if a file is regular."""
+    try:
+        path_stat = path.lstat()
+        return (
+            stat.S_ISREG(path_stat.st_mode)
+            and not stat.S_ISLNK(path_stat.st_mode)
+            and path_stat.st_size > 0
+        )
+    except OSError:
+        return False
+
+
+def is_file_or_symlink(path: Path) -> bool:
+    """Check if a file is regular or symlink."""
+    try:
+        path_stat = path.lstat()
+        return stat.S_ISREG(path_stat.st_mode) or stat.S_ISLNK(path_stat.st_mode)
+    except OSError:
+        return False
+
+
+def is_directory(path: Path) -> bool:
+    """Check if a directory is regular."""
```

Comment:
Do you mean `Check if a path is a real directory.`

Reply:
Yes! fixed in https://github.com/tier4/ota-client/pull/651/commits/2b046db821fc94f132b2bcfa18e2e4cc89892248.